src/LiveLocationManager.ts

```diff
@@ -0,0 +1,256 @@
+      ready: false,
+    });
+    this._deviceId = getDeviceId();
+    this.getDeviceId = getDeviceId;
+    this.watchLocation = watchLocation;
+  }
+
+  public async init() {
+    await this.assureStateInit();
+    this.registerSubscriptions();
+  }
+
+  public registerSubscriptions = () => {
+    this.incrementRefCount();
+    if (this.hasSubscriptions) return;
+
+    this.addUnsubscribeFunction(this.subscribeLiveLocationSharingUpdates());
+    this.addUnsubscribeFunction(this.subscribeTargetMessagesChange());
+  };
+
+  public unregisterSubscriptions = () => super.unregisterSubscriptions();
+
+  get messages() {
+    return this.state.getLatestValue().messages;
+  }
+
+  get stateIsReady() {
+    return this.state.getLatestValue().ready;
+  }
+
+  get deviceId() {
+    if (!this._deviceId) {
+      this._deviceId = this.getDeviceId();
+    }
+    return this._deviceId;
+  }
+
+  private async assureStateInit() {
+    if (this.stateIsReady) return;
+    const { active_live_locations } = await this.client.getSharedLocations();
+    this.state.next({
+      messages: new Map(
+        active_live_locations.map((location) => [location.message_id, location]),
+      ),
+      ready: true,
+    });
+  }
+
+  private subscribeTargetMessagesChange() {
+    let unsubscribeWatchLocation: null | (() => void) = null;
+
+    // Subscribe to location updates only if there are relevant messages to
+    // update, no need for the location watcher to be active/instantiated otherwise
+    const unsubscribe = this.state.subscribeWithSelector(
+      ({ messages }) => ({ messages }),
+      ({ messages }) => {
+        if (!messages.size) {
+          unsubscribeWatchLocation?.();
+          unsubscribeWatchLocation = null;
+        } else if (messages.size && !unsubscribeWatchLocation) {
+          unsubscribeWatchLocation = this.subscribeWatchLocation();
+        }
+      },
+    );
+
+    return () => {
+      unsubscribe();
+      unsubscribeWatchLocation?.();
+    };
+  }
+
+  private subscribeWatchLocation() {
+    let nextAllowedUpdateCallTimestamp = Date.now();
+
+    const unsubscribe = this.watchLocation(({ latitude, longitude }) => {
+      // Integrators can adjust the update interval by supplying custom watchLocation subscription,
+      // but the minimal timeout still has to be set as a failsafe (to prevent rate-limitting)
+      if (Date.now() < nextAllowedUpdateCallTimestamp) return;
+
+      nextAllowedUpdateCallTimestamp =
+        Date.now() + UPDATE_LIVE_LOCATION_REQUEST_MIN_THROTTLE_TIMEOUT;
+
+      withCancellation(LiveLocationManager.symbol, async () => {
+        const promises: Promise<SharedLocationResponse>[] = [];
+        await this.assureStateInit();
+
+        for (const [messageId, location] of this.messages) {
+          if (location.latitude === latitude && location.longitude === longitude)
+            continue;
+          const promise = this.client.updateLocation({
+            created_by_device_id: this.deviceId,
```

Comment:
If the `this.deviceId` is passed here the `updateLocation` API fails with an error as below. I think it should be `location.created_by_device_id`.

<img width="1201" height="349" alt="Screenshot 2025-07-18 at 12 29 55 PM" src="https://github.com/user-attachments/assets/6da81843-8d35-4aa1-885f-da0a095bc1a7" />

Reply:
Yes, you are right Khushal, this should be taken from the location object.